keras/src/layers/layer.py

```diff
@@ -779,6 +783,20 @@ def compute_mask(self, inputs, previous_mask):
     def __call__(self, *args, **kwargs):
         self._check_super_called()
         self._called = True
+        current_remat_mode = get_current_remat_mode()
+
+        if (
+            current_remat_mode != self._remat_mode
+            and current_remat_mode is not None
+        ):
+            warnings.warn(
+                f"The RematScope at call time ({current_remat_mode}) differs "
+                f"the one set during layer initialization "
+                f"({self._remat_mode}). "
+                f"Restoring the correct rematerialization mode "
+                f"{self._remat_mode}for this layer."
+            )
+        # Restore rematerialization mode before calling the layer
```

Comment:
`# Restore rematerialization mode before calling the layer`

Should that comment be line 788? Should it be "verify rematerialization mode"?